build/xtask/src/dist.rs

```diff
@@ -573,6 +575,66 @@ fn write_gdb_script(cfg: &PackageConfig, image_name: &str) -> Result<()> {
     Ok(())
 }
 
+/// Encode the given data as a tagged TLV-C chunk
+fn data_to_tlvc(tag: &str, data: &[u8]) -> Result<Vec<u8>> {
+    if tag.len() != 4 {
+        bail!("Tag must be a 4-byte value, not '{}'", tag);
+    }
+    let mut out = vec![];
+    let mut header = tlvc::ChunkHeader {
+        tag: tag.bytes().collect::<Vec<u8>>().try_into().unwrap(),
+        len: 0.into(),
+        header_checksum: 0.into(),
+    };
+    out.extend(header.as_bytes());
+    let c = tlvc::begin_body_crc();
+    let mut c = c.digest();
+    c.update(data);
+    out.extend(data);
+    let body_len = out.len() - std::mem::size_of::<tlvc::ChunkHeader>();
+    let body_len = u32::try_from(body_len).unwrap();
+    while out.len() & 0b11 != 0 {
+        out.push(0);
+    }
+    out.extend(c.finalize().to_le_bytes());
+
+    // Update the header.
+    header.len.set(body_len);
+    header.header_checksum.set(header.compute_checksum());
+
+    out[..std::mem::size_of::<tlvc::ChunkHeader>()]
+        .copy_from_slice(header.as_bytes());
+    Ok(out)
+}
+
+/// Packs a single blob into a TLV-C structure
+fn pack_blob(blob: &AuxFlashBlob) -> Result<Vec<u8>> {
+    let data = std::fs::read(&blob.file)
+        .context(format!("Could not read blob {}", blob.file))?;
```

Comment:
Nit:

```suggestion
        .with_context(|| format!("Could not read blob {}", blob.file))?;
```

would prevent allocating the error message if the read succeeds.